libr/bin/format/objc/mach0_classes.c

```diff
@@ -0,0 +1,734 @@
+									(32/64-bit pointer) */
+	mach0_ut instanceMethods;			/* SMethodList* (32/64-bit pointer) */
+	mach0_ut classMethods;			/* SMethodList* (32/64-bit pointer) */
+	mach0_ut optionalInstanceMethods;	/* SMethodList* (32/64-bit pointer) */
+	mach0_ut optionalClassMethods;	/* SMethodList* (32/64-bit pointer) */
+	mach0_ut instanceProperties;		/* struct SObjcPropertyList*
+									(32/64-bit pointer) */
+};
+
+Mach0_struct_(SIVarList) {
+	ut32 entsize;
+	ut32 count;
+	/* SIVar first;  These structures follow inline */
+};
+
+Mach0_struct_(SIVar) {
+	mach0_ut offset;		/* uintptr_t * (32/64-bit pointer) */
+	mach0_ut name;		/* const char * (32/64-bit pointer) */
+	mach0_ut type;		/* const char * (32/64-bit pointer) */
+	ut32 alignment;
+	ut32 size;
+};
+
+Mach0_struct_(SObjcProperty) {
+	mach0_ut name;		/* const char * (32/64-bit pointer) */
+	mach0_ut attributes;	/* const char * (32/64-bit pointer) */
+};
+
+Mach0_struct_(SObjcPropertyList) {
+	ut32 entsize;
+	ut32 count;
+	/* struct SObjcProperty first;  These structures follow inline */
+};
+
+///////////////////////////////////////////////////////////////////////////////
+static mach0_ut Mach0_class_(get_pointer)(mach0_ut p, ut32 *offset, ut32 *left,
+									RBinFile *arch);
+
+static void Mach0_class_(copy_sym_name_with_namespace)(char *class_name,
+													char *read_name,
+													RBinSymbol *sym);
+
+static void Mach0_class_(get_ivar_list_t)(mach0_ut p, RBinFile *arch,
+										RBinClass *processed_class);
+
+static void Mach0_class_(get_objc_property_list)(mach0_ut p, RBinFile *arch,
+												RBinClass *processed_class);
+
+static void Mach0_class_(get_method_list_t)(mach0_ut p, RBinFile *arch,
+											char *class_name,
+											RBinClass *processed_class);
+
+static void Mach0_class_(get_protocol_list_t)(mach0_ut p, RBinFile *arch,
+											RBinClass *processed_class);
+
+static void Mach0_class_(get_class_ro_t)(mach0_ut p, RBinFile *arch,
+										ut32 *is_meta_class,
+										RBinClass *processed_class);
+
+static void Mach0_class_(get_class_t)(mach0_ut p, RBinFile *arch,
+									RBinClass *processed_class);
+
+static void Mach0_class_(__r_bin_class_free)(RBinClass *p);
+
+///////////////////////////////////////////////////////////////////////////////
+static mach0_ut Mach0_class_(get_pointer)(mach0_ut p, ut32 *offset, ut32 *left,
+							RBinFile *arch)
+{
+	mach0_ut r;
+	mach0_ut addr;
+	mach0_ut section_content_addr;
+
+	RList *sctns = NULL;
+	RListIter *iter = NULL;
+	RBinSection *s = NULL;
+
+	sctns = r_bin_plugin_mach.sections(arch);
+	if (sctns == NULL) {
+		eprintf("there is no sections\n");
+		return 0;
+	}
+
+	addr = p;
+	r_list_foreach(sctns, iter, s) {
+		if (addr >= s->vaddr && addr < s->vaddr + s->vsize) {
+			if (offset != NULL) {
+				*offset = addr - s->vaddr;
+			}
+			if (left != NULL) {
+				*left = s->vsize - (addr - s->vaddr);
+			}
+
+			//		if(sections[i].protected == TRUE && sections[i].cstring == TRUE)
```

Comment:
Why is it commented? Can we remove it?
If it's a TODO, just put a comment to say so.